app/services/courses/age_range_calculator_service.rb

```diff
@@ -0,0 +1,35 @@
+class Courses::AgeRangeCalculatorService
+  def self.determine_age_range_from(course)
+    new().calculate(course)
+  end
+
+  def calculate(course)
+    return calculate_using_level(course) if course.age_range_in_years.nil?
+
+    calculate_using_age_range_in_years(course)
+  end
+
+private
+
+  def calculate_using_level(course)
+    if %w[primary].include?(course.level)
+      "P"
```

Comment:
Would it be useful to extract these strings ("P", "S" etc) to constants?

Reply:
I feel like they only make sense to UCAS, and only here. My feeling is we don't ... pollute the rest of the code with it. :)